pyaedt/circuit.py

```diff
@@ -1431,6 +1431,90 @@ def assign_power_sinusoidal_excitation_to_ports(self, ports, settings):
 
         return True
 
+    @pyaedt_function_handler()
+    def assign_voltage_frequency_dependent_excitation_to_ports(self, ports, filepath):
+        """Assign a frequency dependent excitation to circuit ports from a frequency dependent source (.fds format).
+
+        Parameters
+        ----------
+        ports : list
+            List of circuit ports to assign to the frequency dependent excitation.
+        filepath : str
+            Path to the frequency dependent file.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+
+        >>> oDesign.UpdateSources
+        """
+        id = self.modeler.schematic.create_unique_id()
+
+        arg1 = [
+            "NAME:NexximSources",
+            [
+                "NAME:NexximSources",
+                [
+                    "NAME:Data",
+                    [
+                        "NAME:VoltageSourceFreq" + str(id),
+                        "DataId:=",
+                        "Source" + str(id),
+                        "Type:=",
+                        17,
+                        "Output:=",
+                        0,
+                        "NumPins:=",
+                        2,
```

Comment:
@Samuelopez-ansys Will this value for the Pins always be `2`?

Reply:
@MaxJPRey  this is a good question. If you check the equivalent methods : assign_current_sinusoidal_excitation_to_ports and assign_voltage_sinusoidal_excitation_to_ports, you will see how NumPins is 2 as well. I does not have any 'sense' because we are not defining any component it is just an excitation, so I think it is always 2 hardcoded. Let's keep it, because for now it is working in these other methods.

Reply:
Make sense. Thanks @Samuelopez-ansys .